Templates/integration_template.ps1

```diff
@@ -41,50 +73,75 @@ if (($env:PSModulePath).Split(';') -ccontains $pwd.Path)
     $env:PSModulePath = ($env:PSModulePath -split ';' | Where-Object {$_ -ne $pwd.path}) -join ';'
 }
 
+# Preserve and set the execution policy so that the DSC MOF can be created
+$executionPolicy = Get-ExecutionPolicy
+if ($executionPolicy -ne 'Unrestricted')
+{
+    Set-ExecutionPolicy -ExecutionPolicy Unrestricted -Force
+    $rollbackExecution = $true
+}
+#endregion
+
+# TODO: Other Init Code Goes Here...
+
 # Using try/finally to always cleanup even if something awful happens.
 try
 {
+
+####################################################################################################
+
     <#
       This file exists so we can load the test file without necessarily having xNetworking in
       the $env:PSModulePath. Otherwise PowerShell will throw an error when reading the Pester File.
     #>
-    $fileName = "$DSCResourceName.ps1"
-    . $PSScriptRoot\$fileName
-    Describe "$DSCResourceName_Integration" {
+    $ConfigFile = Join-Path -Path $PSScriptRoot -ChildPath "$DSCResourceName.Config.ps1"
+    . $ConfigFile
+    
+    Describe "$($DSCResourceName)_Integration" {
+        #region DEFAULT TESTS
         It 'Should compile without throwing' {
             {
                 [System.Environment]::SetEnvironmentVariable('PSModulePath',
                     $env:PSModulePath,[System.EnvironmentVariableTarget]::Machine)
-                $DSCResourceName -OutputPath $env:Temp\$DSCResourceName
-                Start-DscConfiguration -Path $env:Temp\$DSCResourceName -ComputerName localhost -Wait -Verbose -Force
+                Invoke-Expression -Command "$($DSCResourceName)_Config -OutputPath `$WorkingFolder"
+                Start-DscConfiguration -Path (Join-Path -Path $env:Temp -ChildPath $DSCResourceName) `
+                    -ComputerName localhost -Wait -Verbose -Force
             } | Should not throw
         }
 
         It 'should be able to call Get-DscConfiguration without throwing' {
-            {Get-DscConfiguration} | Should Not throw
+            { Get-DscConfiguration -Verbose -ErrorAction Stop } | Should Not throw
         }
+        #endregion
 
         It 'Should have set the firewall and all the parameters should match' {
```

Comment:
This should probably say 'Should have set the resource and all the parameters should match' to be more generic.